### PR TITLE
Add Sentry Integration

### LINKED ISSUE #20: Implement log collecting
As Atlas heads into the phase where it will be finally used outside of the development team, we should integrate some log collecting solution into it. This ensures that if users run into some bugs/errors we don't encounter ourselves, we know of their existence and hopefully can reproduce&fix them.

### PR DESCRIPTION
**Pull request by fcescob**

### CODE REVIEW

File: src/config/urls.ts [hunk 1]
```diff
@@ -4,9 +4,12 @@ const readEnv = (name: string): string | undefined => {
 
 const envQueryNodeUrl = readEnv('REACT_APP_QUERY_NODE_URL')
 const envOrionUrl = readEnv('REACT_APP_ORION_URL')
+const envSentryDNS = readEnv('REACT_APP_SENTRY_DNS')
```
Comment (kdembler) [inline src/config/urls.ts]: I think both the env var and constant should be `DSN` instead?

File: src/mocking/server/index.ts [hunk 2]
```diff
@@ -66,6 +66,11 @@ createServer({
     this.passthrough((request) => {
       return request.url.includes('hotjar')
     })
+
+    // allow Sentry error reporting
```
Comment (kdembler) [inline src/mocking/server/index.ts]: We could probably merge that with the hotjar one, just include additional condition there

File: src/components/LayoutWithRouting.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import styled from '@emotion/styled'
 import { RouteComponentProps, Router, navigate } from '@reach/router'
-import { ErrorBoundary } from 'react-error-boundary'
+import * as Sentry from '@sentry/react'
```
Comment (kdembler) [inline src/components/LayoutWithRouting.tsx]: Why not `import { ErrorBoundary } from '@sentry/react'` instead? That's the syntax we use everywhere

File: src/components/ViewErrorFallback.tsx [hunk 4]
```diff
@@ -34,7 +34,8 @@ const Title = styled(Typography)`
   line-height: 1.25;
 `
 
-const ErrorFallback: React.FC<FallbackProps> = ({ error, resetErrorBoundary }) => {
+const ErrorFallback: FallbackRender = ({ error, componentStack, resetError }) => {
+  console.error(`An error occured in ${componentStack}`)
```
Comment (kdembler) [inline src/components/ViewErrorFallback.tsx]: I guess every `error` logged will be a separate event in Sentry. Maybe it'd make sense to join both of those so we always know which stack links to each error? Something like this?
```js
console.error('An error occurred', {stack: componentStack, error}')
```

Comment (kdembler) [reply src/components/ViewErrorFallback.tsx]: Also `occured` -> `occurred`

File: src/views/HomeView.tsx [hunk 5]
```diff
@@ -55,7 +55,7 @@ const HomeView: React.FC<RouteComponentProps> = () => {
         {!hasNewestVideosError ? (
           <VideoGallery title="Newest videos" loading={newestVideosLoading} videos={newestVideos} />
         ) : (
-          <ErrorFallback error={newestVideosError} resetErrorBoundary={() => refetchNewestVideos()} />
+          <ErrorFallback error={newestVideosError as Error} resetError={() => refetchNewestVideos()} />
```
Comment (kdembler) [inline src/views/HomeView.tsx]: I don't think `as Error` is needed cause `ApolloError` extends that type?

File: src/components/ViewErrorFallback.tsx (hunk 4 above)
Comment (fcescob) [reply src/components/ViewErrorFallback.tsx]: Thanks, changed that

**Review by kdembler — APPROVED** [review]
LGTM

**Pull request merged**